test/functional/p2p_invalid_messages.py

```diff
@@ -134,11 +135,21 @@ def run_test(self):
             # Unmodified message should submit okay.
             node.p2p.send_and_ping(msg)
 
+            # ==elucidate the confusion below==
+            #print(len(node.p2p.build_message(msg))) # 125
+            # => Payload size = 125 - 24 = 101
+            # If we take 77 bytes, then there are 101 - 77 = 24 left
+            # That's exactly the size of a header
+            # So, bitcoind deserializes the header and rejects it for some other reason
+            # But, if we take 78 bytes, then there are 101 - 78 = 23 left
+            # That's not enough to fill a header, so, what happens?
+
```

Comment:
This seems like a very interesting observation, especially for new reviewers. Is it possible to combine the explanation with the question below and elucidate it more clearly? It took me some time to figure out what these statements were referring to.

Reply:
Good point.  This PR is getting a little crowded, I'm going to move the test changes to a new one and I'll clear things up there.